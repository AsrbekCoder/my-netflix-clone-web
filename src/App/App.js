import React from 'react';
import { ReactCreateElement } from '../Components/ReactCreateElement';
import ReactComponent from '../Components/ReactComponent';
import ReactPureComponent from '../Components/ReactPureComponent';
import { FunctionalComponent } from '../Components/FunctionalComponent';

function App() {
  return (
    <>
      <h1>Hello React!</h1>

      <ReactCreateElement />

      <ReactComponent />

      <ReactPureComponent />

      <FunctionalComponent />
    </>
  );
}

export default App;
