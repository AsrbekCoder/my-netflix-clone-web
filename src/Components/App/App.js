import Styles from './App.module.scss';
import reactImg from '../../../public/assets/react.png';

const App = () => {
  return (
    <div className={Styles.container}>
      Hello React!

      <img src={reactImg} alt="some" />
    </div>
  )
};

export default App;