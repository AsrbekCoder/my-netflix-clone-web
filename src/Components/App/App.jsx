import Styles from './App.module.scss';

console.log('one');
var ww = process.env.NODE_ENV.trim.toLowerCase();
if (ww === 'production') {
  console.log('Another one');
} else {
  console.log('test');
}

const App = () => <div className={Styles.container}>Hello React!</div>;

export default App;
