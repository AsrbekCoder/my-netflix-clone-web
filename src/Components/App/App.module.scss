$fontSize: 75px;

.container {
  font-size: $fontSize;
}
