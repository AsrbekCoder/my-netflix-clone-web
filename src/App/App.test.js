import React from 'react';
import { render } from '@testing-library/react';

import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import App from './App';
import {
  DEFAULT_SORT_TYPE_QUERY,
  MOVIES_PAGE_LIMIT,
  MOVIES_PAGE_OFFSET,
  MOVIES_SEARCH_BY_TITLE,
  MOVIES_SORT_ORDER_DESC,
} from '../Utils/Constants';

// eslint-disable-next-line no-undef
describe('Testing App', () => {
  const initialState = {
    moviesData: {
      movies: [],
      isLoading: true,
      moviesRequestBody: {
        filter: '',
        search: '',
        limit: MOVIES_PAGE_LIMIT,
        offset: MOVIES_PAGE_OFFSET,
        sortBy: DEFAULT_SORT_TYPE_QUERY,
        searchBy: MOVIES_SEARCH_BY_TITLE,
        sortOrder: MOVIES_SORT_ORDER_DESC,
      },
    },
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const store = mockStore(initialState);

  const Wrapper = () => (
    <Provider store={store}>
      <App />
    </Provider>
  );

  // eslint-disable-next-line no-undef
  test('Render App without crashing', () => {
    const { asFragment } = render(Wrapper);
    // eslint-disable-next-line no-undef
    expect(asFragment()).toMatchSnapshot();
  });
});
