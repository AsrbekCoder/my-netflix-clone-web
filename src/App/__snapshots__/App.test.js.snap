// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Render App without crashing 1`] = `<DocumentFragment />`;
